manon/table-checkbox.scss

```diff
@@ -27,19 +27,11 @@
 }
 
 table {
+  /* Table within form */
   @extend %table-checkbox;
-}
 
-/* Form within table */
-table {
+  /* Form within table */
   form {
     @extend %table-checkbox;
   }
 }
-
-/* Table within form */
-form {
-  table {
-    @extend %table-checkbox;
-  }
-}
```

Comment:
I'm unsure if the "form within table" scenario on the test page is supposed to be this visually different from the "table within form" scenario:

<img width="315" alt="Screenshot of forms within a table, with the forms having a different background color and large paddings" src="https://github.com/minvws/nl-rdo-manon/assets/67802/876fad86-a19c-4eb1-a413-1082e0816009">

...but the same is true on `main` - this change doesn't affect it.